rswl-cuts-optimization.py

```diff
@@ -0,0 +1,175 @@
+import numpy as np
+from optparse import OptionParser
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+from scipy.stats import binned_statistic
+from scipy import interpolate
+from shower_geometry import impact_parameter
+import rswl_plot
+
+
+def rswl(impact_parameter, size, width, length, rsw_lookup, rsl_lookup):
+
+    wi = interpolate.griddata(
+         (rsw_lookup['impact'], rsw_lookup['size']),
+         rsw_lookup['mean'], (impact_parameter, size), method='linear'
+         )
+
+    sw = interpolate.griddata(
+        (rsw_lookup['impact'], rsw_lookup['size']),
+        rsw_lookup['std'], (impact_parameter, size), method='linear'
+        )
+
+    le = interpolate.griddata(
+         (rsl_lookup['impact'], rsl_lookup['size']),
+         rsl_lookup['mean'], (impact_parameter, size),
+         method='linear'
+         )
+
+    sl = interpolate.griddata(
+        (rsl_lookup['impact'], rsl_lookup['size']),
+        rsl_lookup['std'], (impact_parameter, size),
+        method='linear')
+
+    rsw = (width - wi) / sw
+    rsw = rsw[~np.isnan(rsw)*~np.isinf(rsw)]
+    rsl = (length - le) / sl
+    rsl = rsl[~np.isnan(rsl)*~np.isinf(rsl)]
+
+    return rsw, rsl
+
+
+def efficiency_comp(gamma_cut, rswl):
+
+    efficiency = []
+
+    for ghc in gamma_cut:
```

Comment:
does this work in replacement (avoiding for loops and taking advantage of numpy) ?

```python
bellow_gamma_cut = rswl < gamma_cut[..., np.newaxis]
efficiency = np.sum(bellow_gamma_cut, axis=0)
efficiency /= rswl.shape[0]
```